File: ios/XiangshangYouth/Core/Services/ReportApi.swift
import Foundation
struct ReportApi {
    let client: ApiClient
    init(client: ApiClient = .shared) { self.client = client }
    func report(studentID: String) async throws -> DiagnosisReport {
        try await client.request(path: "v1/students/\(studentID)/report", type: DiagnosisReport.self)
    }
    func refresh(studentID: String) async throws -> DiagnosisReport {
        try await client.request(path: "v1/students/\(studentID)/report/refresh", method: "POST", type: DiagnosisReport.self)
    }
}

File: android/app/src/main/java/com/xiangshang/youth/core/service/ReportApi.kt
package com.xiangshang.youth.core.service

import com.xiangshang.youth.core.model.DiagnosisReport
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface ReportApi {
    @GET("v1/students/{studentId}/report")
    suspend fun report(@Path("studentId") studentId: String): ApiEnvelope<DiagnosisReport>

    @POST("v1/students/{studentId}/report/refresh")
    suspend fun refresh(@Path("studentId") studentId: String): ApiEnvelope<DiagnosisReport>
}
